lib/chef/resource/user.rb

```diff
@@ -42,7 +42,7 @@ def initialize(name, run_context = nil)
         @force = false
         @non_unique = false
         @supports = {
-          :manage_home => false,
+          :manage_home => true,
```

Comment:
There's fairly good discussion in #604 as to why this has been done, and I think it's the correct thing to do - it actually preserves current behaviour in most cases, since by default login.defs will cause home directories to be created. 
Is it a compat break? By the book, yes. In practice, I suspect not.

More concretely, looking at ubuntu, CREATE_HOME was added to login.defs in 2009, and is true by default - so this doesn't change behaviour for any of our supported versions.

Red Hat, and hence CentOS, have carried a patch since at least CentOS 5 to add CREATE_HOME and set it to true, so out of the box this doesn't change behaviour.